app/locales/en/translations.js

```diff
@@ -222,6 +222,7 @@ export default {
     'noSelectedLearnerGroups': 'There are no selected learner groups',
     'instructors': 'Instructors',
     'instructorsManageTitle': 'Manage Instructors',
+    'confirmObjectiveRemoval': 'Are you sure you want to delete this objective?',
```

Comment:
@saschaben / @dart ajax please translate 'Are you sure you want to delete this objective?'